development/architecture/migration-guide/horizontal-migration/migrate-list.md

```diff
@@ -0,0 +1,254 @@
+---
+title: Cleaning up
+weight: 60
+---
+
+# Migrate list
+
+Once you create your controller,and you make your routing file. We can start to have fun with migration.
+
+## Routing
+We will start with the list. You have to add to the routing file an entry for the list.
+
+Keep our example with TitleController.
+
+Example:
+```yaml
+admin_title_index:
+  path: /
+  methods: [ GET, POST ]
+  defaults:
+    _controller: PrestaShopBundle\Controller\Admin\Configure\ShopParameters\TitleController::indexAction
+    _legacy_controller: AdminGender
+    _legacy_link: AdminGender
+```
+
+You must specify as
+
+- `_controller` the namespace of your controller
+- `_legacy_controller` the name of the legacy controller
+- `_legacy_link` the name of the legacy action (for the list it’s the same as the legacy controller)
+
+This information will be helpful to redirect the legacy link to your controller automatically
+
+## Controller
+Once you have finished your routing, your controller must implements this interface: `PrestaShopBundle\Bridge\AdminController\LegacyListControllerBridgeInterface`.
+
+The previous interface needs these two method:
+* `getPositionIdentifier`: have to return position identifier for legacy list request
+* `getIdentifier`: have to return the identifier for table
+
+Then you have to create the corresponding action.
+
+```php
+public function index(Request $request)
+{
+    ....
+}
+```
+
+**Get an instance of `HelperListConfiguration`**
```

Comment:
Missing some abstract info about the bridges before jumping straight to one of them.
Something like:
`Now its time to start creating the "bridges" (link to related article maybe), which will take care of symfony adaptation to internal legacy services.` (you can probably find better wording, but you get the idea)